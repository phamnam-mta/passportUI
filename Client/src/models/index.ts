export type SplitPaneSizes = {
    analyzeSplitPaneSize: number[];
    labelSplitPaneSize: number[];
    labelTableSplitPaneSize: number[];
};
