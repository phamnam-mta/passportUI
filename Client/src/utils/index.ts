export const delay = (ms: number) => {
    return new Promise<void>((resolve) => {
        setTimeout(() => {
            resolve();
        }, ms);
    });
};

export const getPixelWidthFromPercent = (widthInPercentage: number): number => {
    const totalWindowWidth = document.documentElement.clientWidth;
    const paneWidthInPixel = (totalWindowWidth / 100) * widthInPercentage;
    return paneWidthInPixel;
};

export const loadCanvasToBlob = (canvas: HTMLCanvasElement): Promise<Blob> => {
    return new Promise((resolve, reject) => {
        canvas.toBlob((blob) => {
            if (blob) {
                resolve(blob);
            } else {
                reject("Failed to create blob for canvas.");
            }
        });
    });
};

export const loadUrlToArrayBuffer = async (url: string): Promise<ArrayBuffer> => {
    return await fetch(url).then((r) => r.arrayBuffer());
};

export const debounce = (func, timeout = 300) => {
    return (...args) => {
        return setTimeout(() => {
            func.apply(this, args);
        }, timeout);
    };
};
