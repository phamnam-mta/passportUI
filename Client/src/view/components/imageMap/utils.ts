import { Feature } from "ol";
import { Extent } from "ol/extent";
import Polygon from "ol/geom/Polygon";
import ImageLayer from "ol/layer/Image";
import Layer from "ol/layer/Layer";
import VectorLayer from "ol/layer/Vector";
import Projection from "ol/proj/Projection";
import ImageSource from "ol/source/ImageStatic";
import VectorSource from "ol/source/Vector";
import { LAYER_NAME } from "./constants";
import { IRegion } from "./contracts";

export type Dimension = {
    width: number;
    height: number;
};

export const degreeToRadians = (degree: number) => {
    // convert degree to radians
    return (degree * Math.PI * 2) / 360;
};

export const makeLayerFilter = (layerName) => {
    return {
        layerFilter: (layer: Layer) => layer.get(LAYER_NAME) === layerName,
    };
};

export const makeVectorLayer = (layerName, options = {}): Layer => {
    return new VectorLayer({
        [LAYER_NAME]: layerName,
        source: new VectorSource(),
        ...options,
    } as any);
};

export const makeImageLayer = (layerName, uri, projection, imageExtent, options = {}): Layer => {
    return new ImageLayer({
        [LAYER_NAME]: layerName,
        source: makeImageSource(uri, projection, imageExtent),
        ...options,
    } as any);
};

export const makeImageSource = (url: string, projection: Projection, imageExtent: Extent) => {
    return new ImageSource({ url, projection, imageExtent });
};

export const convertRegionToFeature = (region: IRegion, imageExtent: Extent, isOcrProposal: boolean = false) => {
    const coordinates: Array<[number, number]> = [];
    const boundingBox = region.id.split(",").map(parseFloat);
    const imageWidth = imageExtent[2] - imageExtent[0];
    const imageHeight = imageExtent[3] - imageExtent[1];
    for (let i = 0; i < boundingBox.length; i += 2) {
        coordinates.push([Math.round(boundingBox[i] * imageWidth), Math.round((1 - boundingBox[i + 1]) * imageHeight)]);
    }
    const feature = new Feature({
        geometry: new Polygon([coordinates]),
    });
    feature.setProperties({
        id: region.id,
        text: region.value,
        highlighted: false,
        isOcrProposal,
    });
    feature.setId(region.id);

    return feature;
};

export const createRegionIdFromPolygon = (polygon: number[], page: number): string => {
    return polygon.join(",") + ":" + page;
};

export const convertToImageMapCoordinates = (polygon: number[], canvasSize: Dimension, documentSize: Dimension) => {
    const { width: canvasWidth, height: canvasHeight } = canvasSize;
    const { width: ocrWidth, height: ocrHeight } = documentSize;

    const coordinates: number[][] = [];
    for (let i = 0; i < polygon.length; i += 2) {
        coordinates.push([(polygon[i] / ocrWidth) * canvasWidth, (1 - polygon[i + 1] / ocrHeight) * canvasHeight]);
    }
    return coordinates;
};

export const getBoundingBoxFromFeatureId = (id) => id.split(",").map(parseFloat);
