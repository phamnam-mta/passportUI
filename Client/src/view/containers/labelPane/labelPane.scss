@import "~@fluentui/react/dist/sass/References";

.label-pane {
    .label-pane-command-bar {
        height: 44px;
        border-bottom: 1px solid $ms-color-gray30;
    }

    .table-pane {
        .table-pane-header {
            height: 44px;
            border-bottom: 1px solid $ms-color-gray30;

            .table-pane-title {
                max-width: 350px;
                text-overflow: ellipsis;
                overflow: hidden;
                padding-left: 12px;
            }

            .table-pane-type {
                color: $ms-color-gray80;
            }
        }

        .table-label-item-container {
            height: 0;
            margin: 20px;
        }
    }
}
