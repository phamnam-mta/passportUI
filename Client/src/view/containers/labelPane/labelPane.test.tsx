import * as React from "react";
import { FieldType, HeaderType, TableType } from "models/customModels";
import {
    mockAddedField,
    mockDocument,
    mockDocuments,
    mockDocumentLabels,
    mockFields,
    mockDefinitions,
    mockDynamicTableField,
    mockFixedRowTableField,
} from "utils/test";
import { LabelPane } from "./labelPane";
import { FieldLocation } from "store/customModel/customModel";
import { encodeLabelString } from "utils/customModel";
import { shallow } from "enzyme";

describe("<LabelPane />", () => {
    let baseProps;
    const mockFieldKey = "mock key";

    beforeEach(() => {
        baseProps = {
            isTablePaneOpen: false,
            currentDocument: mockDocument,
            fields: mockFields,
            labels: {
                [mockDocument.name]: mockDocumentLabels,
            },
            definitions: mockDefinitions,
            addField: jest.fn(),
            addTableField: jest.fn(),
            deleteField: jest.fn(),
            deleteTableField: jest.fn(),
            renameField: jest.fn(),
            renameTableField: jest.fn(),
            insertTableField: jest.fn(),
            switchSubType: jest.fn(),
            deleteLabelByField: jest.fn(),
            deleteLabelByLabel: jest.fn(),
            assignLabel: jest.fn(),
            setHoveredLabelName: jest.fn(),
            setHideInlineLabelMenu: jest.fn(),
            setIsTablePaneOpen: jest.fn(),
        };
    });

    describe("Rendering", () => {
        it("should render correctly", () => {
            const wrapper = shallow(<LabelPane {...baseProps} />);

            expect(wrapper).toMatchSnapshot();
        });

        it("should match snapshot, when there's no current document", () => {
            const props = { ...baseProps, currentDocument: null, fields: [], labels: {} };

            const wrapper = shallow(<LabelPane {...props} />);

            expect(wrapper).toMatchSnapshot();
        });

        it("should match snapshot, when showAllFields is false", () => {
            const wrapper = shallow(<LabelPane {...baseProps} />);
            wrapper.setState({ showAllFields: false });

            expect(wrapper).toMatchSnapshot();
        });

        it("should match snapshot, when FieldCallout is open", () => {
            const wrapper = shallow(<LabelPane {...baseProps} />);
            wrapper.setState({ isFieldCalloutOpen: true });

            expect(wrapper).toMatchSnapshot();
        });

        it("should match snapshot, when table pane is open with fieldType array", () => {
            const props = { ...baseProps, isTablePaneOpen: true };
            const wrapper = shallow(<LabelPane {...props} />);
            wrapper.setState({ tableFieldKey: mockDynamicTableField.fieldKey });

            expect(wrapper).toMatchSnapshot();
        });

        it("should match snapshot, when table pane is open with fieldType object", () => {
            const props = { ...baseProps, isTablePaneOpen: true };
            const wrapper = shallow(<LabelPane {...props} />);
            wrapper.setState({ tableFieldKey: mockFixedRowTableField.fieldKey });

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("Event Handling", () => {
        it("should clear states, when document is switched", () => {
            const props = { ...baseProps, isTablePaneOpen: true };
            const wrapper = shallow(<LabelPane {...props} />);
            wrapper.setState({
                isFieldCalloutOpen: true,
                createFieldType: FieldType.String,
                tableFieldKey: mockDynamicTableField.fieldKey,
            });
            wrapper.setProps({ currentDocument: mockDocuments[1] });

            expect(baseProps.setHideInlineLabelMenu).toBeCalledTimes(1);
            expect(baseProps.setHideInlineLabelMenu).toBeCalledWith(false);
            expect(baseProps.setIsTablePaneOpen).toBeCalledTimes(1);
            expect(baseProps.setIsTablePaneOpen).toBeCalledWith(false);
            expect(wrapper.state("isFieldCalloutOpen")).toBe(false);
            expect(wrapper.state("createFieldType")).toBeUndefined();
            expect(wrapper.state("tableFieldKey")).toBeUndefined();
        });

        it("should toggle showAllFields, when filter button is clicked", () => {
            const wrapper = shallow(<LabelPane {...baseProps} />);
            const button = wrapper.find("CustomizedIconButton").last();
            button.simulate("click");

            expect(wrapper.state("showAllFields")).toBe(false);
        });

        it.each([
            ["field", 0, FieldType.String],
            ["selectionMark", 1, FieldType.SelectionMark],
            ["signature", 2, FieldType.Signature],
        ])("should open FieldCallout and set create field type, when create %s is clicked", (_, index, type) => {
            const wrapper = shallow(<LabelPane {...baseProps} />);
            const button = wrapper.find("#add-button") as any;
            button.prop("menuProps").items[index].onClick();

            expect(wrapper.state("isFieldCalloutOpen")).toBe(true);
            expect(wrapper.state("createFieldType")).toBe(type);
        });

        it("should handle create field, when FieldCallout onCreateField is triggered", () => {
            const wrapper = shallow(<LabelPane {...baseProps} />);
            wrapper.setState({ isFieldCalloutOpen: true, createFieldType: mockAddedField.fieldType });
            const callout = wrapper.find("FieldCallout") as any;
            callout.prop("onCreateField")(mockAddedField.fieldKey);

            expect(baseProps.addField).toBeCalledTimes(1);
            expect(baseProps.addField).toBeCalledWith(mockAddedField);
        });

        it("should not create field, when FieldCallout onCreateField is triggered with empty value", () => {
            const wrapper = shallow(<LabelPane {...baseProps} />);
            wrapper.setState({ isFieldCalloutOpen: true, createFieldType: mockAddedField.fieldType });
            const callout = wrapper.find("FieldCallout") as any;
            callout.prop("onCreateField")("");

            expect(baseProps.addField).not.toBeCalled();
        });

        it("should handle callout dismiss, when FieldCallout onDismiss is triggered", () => {
            const wrapper = shallow(<LabelPane {...baseProps} />);
            wrapper.setState({ isFieldCalloutOpen: true });
            const callout = wrapper.find("FieldCallout") as any;
            callout.prop("onDismiss")();

            expect(wrapper.state("isFieldCalloutOpen")).toBe(false);
            expect(wrapper.state("createFieldType")).toBeUndefined();
        });

        it("should return exist error, when FieldCallout onGetErrorMessage is triggered with existing field", () => {
            const wrapper = shallow(<LabelPane {...baseProps} />);
            wrapper.setState({ isFieldCalloutOpen: true });
            const callout = wrapper.find("FieldCallout") as any;
            const error = callout.prop("onGetErrorMessage")(mockFields[0].fieldKey);

            expect(error).toBe("The field already exists.");
        });

        it("should not return error, when FieldCallout onGetErrorMessage is triggered with new field", () => {
            const wrapper = shallow(<LabelPane {...baseProps} />);
            wrapper.setState({ isFieldCalloutOpen: true });
            const callout = wrapper.find("FieldCallout") as any;
            const error = callout.prop("onGetErrorMessage")("Non-existing-field");

            expect(error).toBeUndefined();
        });

        it("should handle callout dismiss, when create menu opens", () => {
            const wrapper = shallow(<LabelPane {...baseProps} />);
            const button = wrapper.find("#add-button") as any;
            button.prop("menuProps").onMenuOpened();

            expect(wrapper.state("isFieldCalloutOpen")).toBe(false);
            expect(wrapper.state("createFieldType")).toBeUndefined();
        });

        it("should handle close table pane, when table close button is clicked", () => {
            const props = { ...baseProps, isTablePaneOpen: true };
            const wrapper = shallow(<LabelPane {...props} />);
            wrapper.setState({ tableFieldKey: mockDynamicTableField.fieldKey });
            const button = wrapper.find("#table-close-button");
            button.simulate("click");

            expect(baseProps.setHideInlineLabelMenu).toBeCalledTimes(1);
            expect(baseProps.setHideInlineLabelMenu).toBeCalledWith(false);
            expect(baseProps.setIsTablePaneOpen).toBeCalledTimes(1);
            expect(baseProps.setIsTablePaneOpen).toBeCalledWith(false);
            expect(wrapper.state("tableFieldKey")).toBeUndefined();
        });

        it("should handle delete table field, when TableLabelItem onDeleteField is triggered", () => {
            const mockTableFieldKey = mockDynamicTableField.fieldKey;
            const props = { ...baseProps, isTablePaneOpen: true };
            const wrapper = shallow(<LabelPane {...props} />);

            wrapper.setState({ tableFieldKey: mockTableFieldKey });
            const tableLabelItem = wrapper.find("Connect(TableLabelItem)").first() as any;
            tableLabelItem.prop("onDeleteField")(mockTableFieldKey, mockFieldKey, FieldLocation.field);

            expect(baseProps.deleteTableField).toBeCalledTimes(1);
            expect(baseProps.deleteTableField).toBeCalledWith({
                tableFieldKey: mockTableFieldKey,
                fieldKey: mockFieldKey,
                fieldLocation: FieldLocation.field,
            });
        });

        it("should handle insert table field, when TableLabelItem onInsertField is triggered", () => {
            const mockTableFieldKey = mockDynamicTableField.fieldKey;
            const index = 2;
            const props = { ...baseProps, isTablePaneOpen: true };

            const wrapper = shallow(<LabelPane {...props} />);
            wrapper.setState({ tableFieldKey: mockTableFieldKey });
            const tablelabelItem = wrapper.find("Connect(TableLabelItem)").first() as any;
            tablelabelItem.prop("onInsertField")(mockTableFieldKey, mockFieldKey, index, FieldLocation.field);

            expect(baseProps.insertTableField).toBeCalledTimes(1);
            expect(baseProps.insertTableField).toBeCalledWith({
                tableFieldKey: mockTableFieldKey,
                fieldKey: mockFieldKey,
                index,
                fieldLocation: FieldLocation.field,
            });
        });

        it("should handle rename table field, when TableLabelItem onRenameField is triggered", () => {
            const newName = "newFieldName";
            const mockTableFieldKey = mockDynamicTableField.fieldKey;
            const props = { ...baseProps, isTablePaneOpen: true };

            const wrapper = shallow(<LabelPane {...props} />);
            wrapper.setState({ tableFieldKey: mockTableFieldKey });
            const tablelabelItem = wrapper.find("Connect(TableLabelItem)").first() as any;
            tablelabelItem.prop("onRenameField")(mockTableFieldKey, mockFieldKey, newName, FieldLocation.field);

            expect(baseProps.renameTableField).toBeCalledTimes(1);
            expect(baseProps.renameTableField).toBeCalledWith({
                tableFieldKey: mockTableFieldKey,
                fieldKey: mockFieldKey,
                newName,
                fieldLocation: FieldLocation.field,
            });
        });

        it("should handle assign label, when TableLabelItem onClickCell is triggered", () => {
            const props = { ...baseProps, isTablePaneOpen: true };
            const wrapper = shallow(<LabelPane {...props} />);

            wrapper.setState({ tableFieldKey: mockDynamicTableField.fieldKey });

            const tableLabelItem = wrapper.find("Connect(TableLabelItem)").first() as any;
            const labelName = encodeLabelString(mockDynamicTableField.fieldKey);
            tableLabelItem.prop("onClickCell")(labelName);

            expect(baseProps.assignLabel).toBeCalledTimes(1);
            expect(baseProps.assignLabel).toBeCalledWith(labelName);
        });

        it("should handle delete label, when TableLabelItem onDeleteLabel is triggered", () => {
            const mockLabel = "mockLabel/1/abc";
            const props = { ...baseProps, isTablePaneOpen: true };
            const wrapper = shallow(<LabelPane {...props} />);
            wrapper.setState({ tableFieldKey: mockDynamicTableField.fieldKey });

            const labelItem = wrapper.find("Connect(TableLabelItem)").first() as any;
            labelItem.prop("onDeleteLabel")(mockLabel);

            expect(baseProps.deleteLabelByLabel).toBeCalledTimes(1);
            expect(baseProps.deleteLabelByLabel).toBeCalledWith(mockLabel);
        });

        it("should open CreateTableModal, when create table is clicked", () => {
            const wrapper = shallow(<LabelPane {...baseProps} />);
            const button = wrapper.find("#add-button") as any;
            button.prop("menuProps").items[3].onClick();

            expect(wrapper.state("isCreateTableModalOpen")).toBe(true);
        });

        it("should close CreateTableModal, when onClose of CreateTableModal is triggered", () => {
            const wrapper = shallow(<LabelPane {...baseProps} />);
            wrapper.setState({ isCreateTableModalOpen: true });
            const modal = wrapper.find("CreateTableModal") as any;
            modal.prop("onClose")();

            expect(wrapper.state("isCreateTableModalOpen")).toBe(false);
        });

        it("should handle create table field, when onCreateField of CreateTableModal is triggered", () => {
            const wrapper = shallow(<LabelPane {...baseProps} />);
            const modal = wrapper.find("CreateTableModal") as any;
            modal.prop("onCreateField")(mockFieldKey, TableType.fixed, HeaderType.column);

            expect(baseProps.addTableField).toBeCalledTimes(1);
            expect(baseProps.addTableField).toBeCalledWith({
                fieldKey: mockFieldKey,
                tableType: TableType.fixed,
                headerType: HeaderType.column,
            });
        });

        it("should return exist error, when LabelItem onGetRenameErrorMessage is triggered with existing field", () => {
            const wrapper = shallow(<LabelPane {...baseProps} />);
            const modal = wrapper.find("CreateTableModal").first() as any;
            const error = modal.prop("onGetNameErrorMessage")(mockFields[0].fieldKey);

            expect(error).toBe("The field already exists.");
        });

        it("should not return error, when LabelItem onGetRenameErrorMessage is triggered with new field", () => {
            const wrapper = shallow(<LabelPane {...baseProps} />);
            const modal = wrapper.find("CreateTableModal").first() as any;
            const error = modal.prop("onGetNameErrorMessage")("Non-existing-field");

            expect(error).toBeUndefined();
        });

        it("should handle open table pane, when table field is clicked", () => {
            const wrapper = shallow(<LabelPane {...baseProps} />);
            const labelItem = wrapper.find("Connect(LabelList)").first() as any;
            labelItem.prop("onTablePaneOpen")(mockDynamicTableField);

            expect(baseProps.setIsTablePaneOpen).toBeCalledTimes(1);
            expect(baseProps.setIsTablePaneOpen).toBeCalledWith(true);
            expect(wrapper.state("tableFieldKey")).toBe(mockDynamicTableField.fieldKey);
        });
    });
});
