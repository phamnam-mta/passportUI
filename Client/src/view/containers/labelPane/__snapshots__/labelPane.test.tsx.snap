// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LabelPane /> Rendering should match snapshot, when FieldCallout is open 1`] = `
<Stack
  className="label-pane"
  grow={1}
>
  <Stack
    className="label-pane-command-bar"
    horizontal={true}
    horizontalAlign="space-between"
    verticalAlign="center"
  >
    <div />
    <Stack
      horizontal={true}
    >
      <CustomizedIconButton
        iconProps={
          Object {
            "iconName": "Add",
          }
        }
        id="add-button"
        menuProps={
          Object {
            "directionalHint": 6,
            "items": Array [
              Object {
                "fieldType": "string",
                "iconProps": Object {
                  "iconName": "GroupList",
                },
                "key": "field",
                "onClick": [Function],
                "text": "Field",
              },
              Object {
                "fieldType": "selectionMark",
                "iconProps": Object {
                  "iconName": "CheckboxComposite",
                },
                "key": "selectionMark",
                "onClick": [Function],
                "text": "Selection Mark",
              },
              Object {
                "fieldType": "signature",
                "iconProps": Object {
                  "iconName": "WhiteBoardApp16",
                },
                "key": "signature",
                "onClick": [Function],
                "text": "Signature",
              },
              Object {
                "iconProps": Object {
                  "iconName": "Table",
                },
                "key": "table",
                "onClick": [Function],
                "text": "Table",
              },
            ],
            "onMenuOpened": [Function],
          }
        }
        styles={
          Object {
            "menuIcon": Object {
              "height": 0,
              "margin": 0,
              "overflow": "hidden",
              "width": 0,
            },
          }
        }
        title="Add field"
      />
      <CustomizedIconButton
        iconProps={
          Object {
            "iconName": "Filter",
          }
        }
        onClick={[Function]}
        styles={
          Object {
            "root": Object {
              "display": "none",
            },
          }
        }
      />
    </Stack>
  </Stack>
  <Stack
    grow={1}
    style={
      Object {
        "height": 0,
      }
    }
  >
    <Connect(LabelList)
      onTablePaneOpen={[Function]}
    />
  </Stack>
  <FieldCallout
    directionalHint={6}
    directionalHintFixed={true}
    onCreateField={[Function]}
    onDismiss={[Function]}
    onGetErrorMessage={[Function]}
    target="#add-button"
    width={268}
  />
  <CreateTableModal
    isOpen={false}
    onClose={[Function]}
    onCreateField={[Function]}
    onGetNameErrorMessage={[Function]}
  />
</Stack>
`;

exports[`<LabelPane /> Rendering should match snapshot, when showAllFields is false 1`] = `
<Stack
  className="label-pane"
  grow={1}
>
  <Stack
    className="label-pane-command-bar"
    horizontal={true}
    horizontalAlign="space-between"
    verticalAlign="center"
  >
    <div />
    <Stack
      horizontal={true}
    >
      <CustomizedIconButton
        iconProps={
          Object {
            "iconName": "Add",
          }
        }
        id="add-button"
        menuProps={
          Object {
            "directionalHint": 6,
            "items": Array [
              Object {
                "fieldType": "string",
                "iconProps": Object {
                  "iconName": "GroupList",
                },
                "key": "field",
                "onClick": [Function],
                "text": "Field",
              },
              Object {
                "fieldType": "selectionMark",
                "iconProps": Object {
                  "iconName": "CheckboxComposite",
                },
                "key": "selectionMark",
                "onClick": [Function],
                "text": "Selection Mark",
              },
              Object {
                "fieldType": "signature",
                "iconProps": Object {
                  "iconName": "WhiteBoardApp16",
                },
                "key": "signature",
                "onClick": [Function],
                "text": "Signature",
              },
              Object {
                "iconProps": Object {
                  "iconName": "Table",
                },
                "key": "table",
                "onClick": [Function],
                "text": "Table",
              },
            ],
            "onMenuOpened": [Function],
          }
        }
        styles={
          Object {
            "menuIcon": Object {
              "height": 0,
              "margin": 0,
              "overflow": "hidden",
              "width": 0,
            },
          }
        }
        title="Add field"
      />
      <CustomizedIconButton
        iconProps={
          Object {
            "iconName": "ClearFilter",
          }
        }
        onClick={[Function]}
        styles={
          Object {
            "root": Object {
              "display": "none",
            },
          }
        }
      />
    </Stack>
  </Stack>
  <Stack
    grow={1}
    style={
      Object {
        "height": 0,
      }
    }
  >
    <Connect(LabelList)
      onTablePaneOpen={[Function]}
    />
  </Stack>
  <CreateTableModal
    isOpen={false}
    onClose={[Function]}
    onCreateField={[Function]}
    onGetNameErrorMessage={[Function]}
  />
</Stack>
`;

exports[`<LabelPane /> Rendering should match snapshot, when table pane is open with fieldType array 1`] = `
<Stack
  className="label-pane"
  grow={1}
>
  <Stack
    className="table-pane"
    grow={1}
  >
    <Stack
      className="table-pane-header"
      horizontal={true}
      horizontalAlign="space-between"
      verticalAlign="center"
    >
      <Stack
        horizontal={true}
        tokens={
          Object {
            "childrenGap": 8,
          }
        }
      >
        <Text
          className="table-pane-title"
        >
          DynamicTable/abc
        </Text>
        <Text
          className="table-pane-type"
        >
          Table
        </Text>
      </Stack>
      <CustomizedIconButton
        iconProps={
          Object {
            "iconName": "Cancel",
          }
        }
        id="table-close-button"
        onClick={[Function]}
        title="Close"
      />
    </Stack>
    <Stack
      grow={1}
    >
      <ol>
        <li
          key="0"
        >
          In your document, select the words you want to label.
        </li>
        <li
          key="1"
        >
          In the labeling pane, click the table cell to assign them to that cell.
        </li>
      </ol>
      <Stack
        className="table-label-item-container"
        grow={1}
      >
        <Connect(TableLabelItem)
          definition={
            Object {
              "fieldFormat": "not-specified",
              "fieldKey": "0b53a58a-3345-37db-e525-2c43c370bbc7",
              "fieldType": "object",
              "fields": Array [
                Object {
                  "fieldFormat": "not-specified",
                  "fieldKey": "CUSTOMER#",
                  "fieldType": "number",
                },
                Object {
                  "fieldFormat": "not-specified",
                  "fieldKey": "LICNESE#",
                  "fieldType": "number",
                },
                Object {
                  "fieldFormat": "not-specified",
                  "fieldKey": "TERMS",
                  "fieldType": "string",
                },
              ],
              "itemType": null,
            }
          }
          field={
            Object {
              "fieldFormat": "not-specified",
              "fieldKey": "DynamicTable/abc",
              "fieldType": "array",
              "fields": null,
              "itemType": "0b53a58a-3345-37db-e525-2c43c370bbc7",
            }
          }
          onClickCell={[Function]}
          onDeleteField={[Function]}
          onDeleteLabel={[Function]}
          onInsertField={[Function]}
          onItemMouseEnter={[Function]}
          onItemMouseLeave={[Function]}
          onRenameField={[Function]}
          tableLabels={
            Object {
              "DynamicTable~1abc/0/CUSTOMER#": Object {
                "label": "DynamicTable~1abc/0/CUSTOMER#",
                "value": Array [
                  Object {
                    "boundingBoxes": Array [
                      Array [
                        0.8563411764705883,
                        0.18608181818181818,
                        0.8621529411764706,
                        0.18608181818181818,
                        0.8621529411764706,
                        0.1929909090909091,
                        0.8563411764705883,
                        0.1929909090909091,
                      ],
                    ],
                    "page": 1,
                    "text": "13267",
                  },
                ],
              },
            }
          }
        />
      </Stack>
    </Stack>
  </Stack>
  <CreateTableModal
    isOpen={false}
    onClose={[Function]}
    onCreateField={[Function]}
    onGetNameErrorMessage={[Function]}
  />
</Stack>
`;

exports[`<LabelPane /> Rendering should match snapshot, when table pane is open with fieldType object 1`] = `
<Stack
  className="label-pane"
  grow={1}
>
  <Stack
    className="table-pane"
    grow={1}
  >
    <Stack
      className="table-pane-header"
      horizontal={true}
      horizontalAlign="space-between"
      verticalAlign="center"
    >
      <Stack
        horizontal={true}
        tokens={
          Object {
            "childrenGap": 8,
          }
        }
      >
        <Text
          className="table-pane-title"
        >
          FixedRowTable
        </Text>
        <Text
          className="table-pane-type"
        >
          Table
        </Text>
      </Stack>
      <CustomizedIconButton
        iconProps={
          Object {
            "iconName": "Cancel",
          }
        }
        id="table-close-button"
        onClick={[Function]}
        title="Close"
      />
    </Stack>
    <Stack
      grow={1}
    >
      <ol>
        <li
          key="0"
        >
          In your document, select the words you want to label.
        </li>
        <li
          key="1"
        >
          In the labeling pane, click the table cell to assign them to that cell.
        </li>
      </ol>
      <Stack
        className="table-label-item-container"
        grow={1}
      >
        <Connect(TableLabelItem)
          definition={
            Object {
              "fieldFormat": "not-specified",
              "fieldKey": "f4009382-976c-1d1d-ca9d-4db42d5d9eac",
              "fieldType": "object",
              "fields": Array [
                Object {
                  "fieldFormat": "not-specified",
                  "fieldKey": "DATE",
                  "fieldType": "date",
                },
                Object {
                  "fieldFormat": "not-specified",
                  "fieldKey": "INVOICE#",
                  "fieldType": "number",
                },
              ],
            }
          }
          field={
            Object {
              "fieldFormat": "not-specified",
              "fieldKey": "FixedRowTable",
              "fieldType": "object",
              "fields": Array [
                Object {
                  "fieldFormat": "not-specified",
                  "fieldKey": "abc",
                  "fieldType": "f4009382-976c-1d1d-ca9d-4db42d5d9eac",
                },
                Object {
                  "fieldFormat": "not-specified",
                  "fieldKey": "def",
                  "fieldType": "cd309d8e-1fd7-335f-e60c-3909c781556a",
                },
              ],
              "itemType": null,
              "visualizationHint": "horizontal",
            }
          }
          onClickCell={[Function]}
          onDeleteField={[Function]}
          onDeleteLabel={[Function]}
          onInsertField={[Function]}
          onItemMouseEnter={[Function]}
          onItemMouseLeave={[Function]}
          onRenameField={[Function]}
          tableLabels={
            Object {
              "FixedRowTable/abc/DATE": Object {
                "label": "FixedRowTable/abc/DATE",
                "value": Array [
                  Object {
                    "boundingBoxes": Array [
                      Array [
                        0.10788235294117647,
                        0.6268090909090909,
                        0.1326,
                        0.6268090909090909,
                        0.1326,
                        0.6343545454545455,
                        0.10788235294117647,
                        0.6343545454545455,
                      ],
                    ],
                    "page": 1,
                    "text": "130",
                  },
                ],
              },
            }
          }
        />
      </Stack>
    </Stack>
  </Stack>
  <CreateTableModal
    isOpen={false}
    onClose={[Function]}
    onCreateField={[Function]}
    onGetNameErrorMessage={[Function]}
  />
</Stack>
`;

exports[`<LabelPane /> Rendering should match snapshot, when there's no current document 1`] = `
<Stack
  className="label-pane"
  grow={1}
>
  <Stack
    className="label-pane-command-bar"
    horizontal={true}
    horizontalAlign="space-between"
    verticalAlign="center"
  >
    <div />
    <Stack
      horizontal={true}
    >
      <CustomizedIconButton
        iconProps={
          Object {
            "iconName": "Add",
          }
        }
        id="add-button"
        menuProps={
          Object {
            "directionalHint": 6,
            "items": Array [
              Object {
                "fieldType": "string",
                "iconProps": Object {
                  "iconName": "GroupList",
                },
                "key": "field",
                "onClick": [Function],
                "text": "Field",
              },
              Object {
                "fieldType": "selectionMark",
                "iconProps": Object {
                  "iconName": "CheckboxComposite",
                },
                "key": "selectionMark",
                "onClick": [Function],
                "text": "Selection Mark",
              },
              Object {
                "fieldType": "signature",
                "iconProps": Object {
                  "iconName": "WhiteBoardApp16",
                },
                "key": "signature",
                "onClick": [Function],
                "text": "Signature",
              },
              Object {
                "iconProps": Object {
                  "iconName": "Table",
                },
                "key": "table",
                "onClick": [Function],
                "text": "Table",
              },
            ],
            "onMenuOpened": [Function],
          }
        }
        styles={
          Object {
            "menuIcon": Object {
              "height": 0,
              "margin": 0,
              "overflow": "hidden",
              "width": 0,
            },
          }
        }
        title="Add field"
      />
      <CustomizedIconButton
        iconProps={
          Object {
            "iconName": "Filter",
          }
        }
        onClick={[Function]}
        styles={
          Object {
            "root": Object {
              "display": "none",
            },
          }
        }
      />
    </Stack>
  </Stack>
  <Stack
    grow={1}
    style={
      Object {
        "height": 0,
      }
    }
  >
    <Connect(LabelList)
      onTablePaneOpen={[Function]}
    />
  </Stack>
  <CreateTableModal
    isOpen={false}
    onClose={[Function]}
    onCreateField={[Function]}
    onGetNameErrorMessage={[Function]}
  />
</Stack>
`;

exports[`<LabelPane /> Rendering should render correctly 1`] = `
<Stack
  className="label-pane"
  grow={1}
>
  <Stack
    className="label-pane-command-bar"
    horizontal={true}
    horizontalAlign="space-between"
    verticalAlign="center"
  >
    <div />
    <Stack
      horizontal={true}
    >
      <CustomizedIconButton
        iconProps={
          Object {
            "iconName": "Add",
          }
        }
        id="add-button"
        menuProps={
          Object {
            "directionalHint": 6,
            "items": Array [
              Object {
                "fieldType": "string",
                "iconProps": Object {
                  "iconName": "GroupList",
                },
                "key": "field",
                "onClick": [Function],
                "text": "Field",
              },
              Object {
                "fieldType": "selectionMark",
                "iconProps": Object {
                  "iconName": "CheckboxComposite",
                },
                "key": "selectionMark",
                "onClick": [Function],
                "text": "Selection Mark",
              },
              Object {
                "fieldType": "signature",
                "iconProps": Object {
                  "iconName": "WhiteBoardApp16",
                },
                "key": "signature",
                "onClick": [Function],
                "text": "Signature",
              },
              Object {
                "iconProps": Object {
                  "iconName": "Table",
                },
                "key": "table",
                "onClick": [Function],
                "text": "Table",
              },
            ],
            "onMenuOpened": [Function],
          }
        }
        styles={
          Object {
            "menuIcon": Object {
              "height": 0,
              "margin": 0,
              "overflow": "hidden",
              "width": 0,
            },
          }
        }
        title="Add field"
      />
      <CustomizedIconButton
        iconProps={
          Object {
            "iconName": "Filter",
          }
        }
        onClick={[Function]}
        styles={
          Object {
            "root": Object {
              "display": "none",
            },
          }
        }
      />
    </Stack>
  </Stack>
  <Stack
    grow={1}
    style={
      Object {
        "height": 0,
      }
    }
  >
    <Connect(LabelList)
      onTablePaneOpen={[Function]}
    />
  </Stack>
  <CreateTableModal
    isOpen={false}
    onClose={[Function]}
    onCreateField={[Function]}
    onGetNameErrorMessage={[Function]}
  />
</Stack>
`;
